src/main/java/se/laz/casual/statistics/EventWriter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+
+package se.laz.casual.statistics;
+
+import java.util.Objects;
+import java.util.function.BooleanSupplier;
+
+public record EventWriter(AugmentedEventStore eventStore, StoreFunction storeFunction, BooleanSupplier condition)
```

Comment:
Add documentation for each class with a one liner explaining it's functionality.